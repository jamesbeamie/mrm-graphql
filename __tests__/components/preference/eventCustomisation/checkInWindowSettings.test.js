import React from 'react';
import { shallow, mount } from 'enzyme';
import { RadioGroup, RadioButton } from 'react-toolbox/lib/radio';
import CheckInWindowSettings from '../../../../src/components/preference/eventCustomisation/checkInWindowSettings';


describe('Tests for CheckInWindowSettings Component', () => {
  const Wrapper = mount(<CheckInWindowSettings />);
  const shallowWrapper = shallow(<CheckInWindowSettings />);

  it('renders correctly from memory', () => {
    expect(shallowWrapper).toMatchSnapshot();
  });

  it('calls handleChange function', () => {
    expect(shallowWrapper.instance().handleChange());
  });

  it('sets value in state when user clicks on one of the time options', () => {
    const radioButton = Wrapper.find(RadioGroup).find(RadioButton).first();
    radioButton.find('input').simulate('click', { target: { checked: true } });
    expect(Wrapper.instance().state.value).toEqual('2');
  });
});
