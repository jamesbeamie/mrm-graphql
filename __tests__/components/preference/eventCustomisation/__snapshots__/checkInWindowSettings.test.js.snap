// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for CheckInWindowSettings Component renders correctly from memory 1`] = `
<Fragment>
  <h1>
    Event Customizations
  </h1>
  <div
    className="notification-container"
  >
    <section
      className="check-in"
    >
      <h2
        className="notification-title"
      >
        Checkin Window
      </h2>
      <p>
        Time before you can check into a meeting
      </p>
    </section>
    <section
      className="checkin-time"
    >
      <ThemedIconMenu
        className="time-picker-dropdown"
        composeTheme="deeply"
        icon={
          <div
            className="time-picker"
          >
            <span>
              10
               minutes
            </span>
          </div>
        }
        iconRipple={true}
        mapThemrProps={[Function]}
        menuRipple={true}
        position="auto"
        selectable={false}
      >
        <ThemedRadioGroup
          className="radio-wrapper"
          composeTheme="deeply"
          disabled={false}
          mapThemrProps={[Function]}
          onChange={[Function]}
          value="10"
        >
          <ThemedRadioButton
            checked={false}
            className=""
            composeTheme="deeply"
            disabled={false}
            label="2 mins"
            mapThemrProps={[Function]}
            value="2"
          />
          <ThemedRadioButton
            checked={false}
            className=""
            composeTheme="deeply"
            disabled={false}
            label="5 mins"
            mapThemrProps={[Function]}
            value="5"
          />
          <ThemedRadioButton
            checked={false}
            className=""
            composeTheme="deeply"
            disabled={false}
            label="10 mins"
            mapThemrProps={[Function]}
            value="10"
          />
          <ThemedRadioButton
            checked={false}
            className=""
            composeTheme="deeply"
            disabled={false}
            label="15 mins"
            mapThemrProps={[Function]}
            value="15"
          />
        </ThemedRadioGroup>
      </ThemedIconMenu>
    </section>
  </div>
</Fragment>
`;
